wip: migrate to opentelemetry (#10223)
traefik/traefik: docs/content/migration/v2-to-v3.md

In v3, the Traefik CRD API Version `apiextensions.k8s.io/v1beta1` ([removed since Kubernetes v1.22](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122)) support has been removed.

Please use the CRD definition with the API Version `apiextensions.k8s.io/v1` instead.

## Tracing

> In v3, the tracing feature has been revamped and is now powered exclusively by OpenTelemetry (OTel).

Reviewer: ```suggestion
In v3, the tracing feature has been revamped and is now powered exclusively by [OpenTelemetry](https://opentelemetry.io/ "Link to website of OTel") (OTel).
```

> It's important to note that Traefik v3 no longer supports direct output formats for specific vendors such as Instana, Jaeger, Zipkin, Haystack, Datadog, and Elastic.

Reviewer: ```suggestion
!!! warning "Important"

    Traefik v3 **no** longer supports direct output formats for specific vendors such as Instana, Jaeger, Zipkin, Haystack, Datadog, and Elastic.
```
